src/assert.js

```diff
@@ -15,25 +15,38 @@ QUnit.assert = Assert.prototype = {
 		}
 	},
 
-	// Increment this Test's semaphore counter, then return a single-use function that
+	// Increment this Test's semaphore counter, then return a function that
 	// decrements that counter a maximum of once.
-	async: function() {
+	async: function( count ) {
 		var test = this.test,
-			popped = false;
+			popped = false,
+			acceptCallCount = count;
+
+		if ( typeof acceptCallCount === "undefined" ) {
+			acceptCallCount = 1;
+		}
 
 		test.semaphore += 1;
 		test.usedAsync = true;
 		pauseProcessing();
 
 		return function done() {
-			if ( !popped ) {
-				test.semaphore -= 1;
-				popped = true;
-				resumeProcessing();
-			} else {
-				test.pushFailure( "Called the callback returned from `assert.async` more than once",
+			acceptCallCount -= 1;
+
+			if ( acceptCallCount > 0 ) {
+				pauseProcessing();
```

Comment:
Assuming that's the case (which it seems to be), the function might read better with a little more rearranging:
```js
if ( popped ) {
	// push failure and return
}

acceptCallCount -= 1;
if ( acceptCallCount === 0 ) {
	// set popped and resume processing
}
```